Collection.js

```diff
@@ -12,26 +12,29 @@ Invoices.byTimeRange = function(filter, sortBy, sortOrder,limit){
   }
   sortQuery[sortBy] = sortOrder;
 
-  let curDate = new Date();
+  let endDate = new Date();
   let startDate = new Date();
+  let daywrapper = moment(startDate);
   switch (filter) {
     default:
     case "today":
-      startDate.setDate(curDate.getDate()-1);
+     daywrapper.startOf('day'); 
       break;
     case "week":
-      startDate.setDate(curDate.getDate()-7);
+      daywrapper.startOf('week');
+      daywrapper.add(1, 'days');//begin on mondays
       break;
     case "month":
-      startDate.setMonth(curDate.getMonth()-1);
+      daywrapper.startOf('month'); 
       break;
     case "all":
-      return Invoices.find({}, {sort: sortQuery,limit:limit});
+      return Invoices.find({}, 
+        {sort: sortQuery,limit:limit});
       break;
   }
   return Invoices.find(
-    {createdAt: {$gte: startDate}},
-    { sort: sortQuery,limit:limit }
+    {createdAt: {$gte: daywrapper.toDate(),$lte:  endDate}},
```

Comment:
You `endDate` is just the current date! This is not what you want here … it should be the end of the current day / week / month :wink: checkout `moment.endOf(…)` it works the same as `startOf`